solidity/test/avs/HyperlaneServiceManager.t.sol

```diff
@@ -0,0 +1,470 @@
+            numQueued
+        );
+        for (uint8 i = 0; i < numQueued; i++) {
+            queuedChallengers[i] = challengers[i];
+        }
+        IRemoteChallenger[]
+            memory unqueuedChallengers = new IRemoteChallenger[](
+                numOfChallengers - numQueued
+            );
+        for (uint8 i = numQueued; i < numOfChallengers; i++) {
+            unqueuedChallengers[i - numQueued] = challengers[i];
+        }
+
+        vm.startPrank(operator);
+        _hsm.enrollIntoChallengers(challengers);
+        _assertChallengers(challengers, EnrollmentStatus.ENROLLED, 0);
+
+        _hsm.startUnenrollment(queuedChallengers);
+        _assertChallengers(
+            queuedChallengers,
+            EnrollmentStatus.PENDING_UNENROLLMENT,
+            block.number
+        );
+        _assertChallengers(unqueuedChallengers, EnrollmentStatus.ENROLLED, 0);
+
+        vm.stopPrank();
+    }
+
+    /// forge-config: default.fuzz.runs = 10
+    function testFuzz_completeQueuedUnenrollmentFromChallenger(
+        uint8 numOfChallengers,
+        uint8 numUnenrollable
+    ) public {
+        vm.assume(numUnenrollable > 0 && numUnenrollable <= numOfChallengers);
+
+        _registerOperator();
+        IRemoteChallenger[] memory challengers = _deployChallengers(
+            numOfChallengers
+        );
+        IRemoteChallenger[]
+            memory unenrollableChallengers = new IRemoteChallenger[](
+                numUnenrollable
+            );
+        for (uint8 i = 0; i < numUnenrollable; i++) {
+            unenrollableChallengers[i] = challengers[i];
+        }
+
+        vm.startPrank(operator);
+        _hsm.enrollIntoChallengers(challengers);
+        _hsm.startUnenrollment(challengers);
+
+        _assertChallengers(
+            challengers,
+            EnrollmentStatus.PENDING_UNENROLLMENT,
+            block.number
+        );
+
+        vm.expectRevert();
+        _hsm.completeUnenrollment(unenrollableChallengers);
+
+        vm.roll(block.number + challengeDelayBlocks);
+
+        _hsm.completeUnenrollment(unenrollableChallengers);
+
+        assertEq(
+            _hsm.getOperatorChallengers(operator).length,
+            numOfChallengers - numUnenrollable
+        );
+
+        vm.stopPrank();
+    }
+
+    /// forge-config: default.fuzz.runs = 10
+    function testFuzz_freezeOperator(uint8 numOfChallengers) public {
+        _registerOperator();
+
+        IRemoteChallenger[] memory challengers = _deployChallengers(
+            numOfChallengers
+        );
+
+        vm.prank(operator);
+        _hsm.enrollIntoChallengers(challengers);
+
+        for (uint256 i = 0; i < challengers.length; i++) {
+            vm.expectCall(
+                address(slasher),
+                abi.encodeCall(ISlasher.freezeOperator, (operator))
+            );
+            challengers[i].handleChallenge(operator);
+        }
+    }
+
+    /// forge-config: default.fuzz.runs = 10
```

Comment:
should we just configure this in the foundry.toml? whats the default number of fuzz runs?

Reply:
256 I believe but for local testing it was too slow. I can remove it.

Reply:
switched to global fuzz.runs = 50 and higher dictionary weight which means it will looks for edge case values more frequently here https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/3651/commits/23aa15529a22408b9baef54bff151b484f249937